tests/phpunit/CRM/Mailing/BaseMailingSystemTest.php

```diff
@@ -0,0 +1,407 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2016                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+ */
+
+/**
+ * Test that content produced by CiviMail looks the way it's expected.
+ *
+ * @package CiviCRM_APIv3
+ * @subpackage API_Job
+ *
+ * @copyright CiviCRM LLC (c) 2004-2016
+ * @version $Id: Job.php 30879 2010-11-22 15:45:55Z shot $
+ *
+ */
+
+/**
+ * Class CRM_Mailing_MailingSystemTest
+ * @group headless
+ * @see \Civi\FlexMailer\FlexMailerSystemTest
+ * @see CRM_Mailing_MailingSystemTest
+ */
+abstract class CRM_Mailing_BaseMailingSystemTest extends CiviUnitTestCase {
```

Comment:
It's true that no `function` is specifically `abstract`, so syntactically there's no obligation to mark the class `abstract`. 

However, semantically, the class is abstract. There will be two subclasses (`MailingSystemTest` and `FlexMailerSystemTest`) which extend `BaseMailingSystemTest`.  `BaseMailingSystemTest` does not do everything needed for a test. It's abstract because
 * We don't want `phpunit` to execute `BaseMailingSystemTest` by itself. (It's not a complete test.)
 * Subclasses are expected to make some changes -- by overriding `setUp()`.